### PR TITLE
[O2B-1274] Fetching GAQ detectors

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [ ] issue has "Fix version" assigned
- [ ] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Added endpoint for fetching GAQ detectors

Changes made to the database:
- NA

### CODE REVIEW

File: lib/server/controllers/dplDetectors.controller.js [hunk 1]
```diff
@@ -26,3 +29,31 @@ exports.listAllDplDetectorsHandler = async (_request, response) => {
...
+        })),
+        req,
+        res,
+    );
+    if (validatedDTO) {
+        try {
+            const {
+                dataPassId,
+                runNumber,
+            } = validatedDTO.query;
```
Comment (martinboulais) [inline lib/server/controllers/dplDetectors.controller.js:51]: Oneline

File: lib/server/services/dpl/DplDetectorsService.js [hunk 2]
```diff
@@ -26,6 +26,20 @@ class DplDetectorsService {
...
+     *
+     * @param {number} dataPassId id of data pass
+     * @param {number} runNumber run number
+     * @return {Promise<Partial<DplDetector>[]>} promise of DPL detectors
+     */
+    async getGaqDetectors(dataPassId, runNumber) {
+        const detectors = await DplDetectorRepository.findAll({
+            include: [{ association: 'gaqAssociation', where: { dataPassId, runNumber } }],
+        });
+        return detectors.map(dplDetectorAdapter.toMinifiedEntity);
```
Comment (martinboulais) [inline lib/server/services/dpl/DplDetectorsService.js:41]: As said previously this is an error, this should be in something like `gaqService`. Same applies for router, controller

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/server/controllers/dplDetectors.controller.js (hunk 1 above)
Comment (xsalonx) [reply lib/server/controllers/dplDetectors.controller.js:51]: applied

File: lib/database/models/dpl/dpldetector.js [hunk 3]
```diff
@@ -29,6 +29,7 @@ module.exports = (sequelize) => {
 
     DplDetector.associate = (models) => {
         DplDetector.hasMany(models.DplProcessExecution, { as: 'processesExecutions', foreignKey: 'detectorId' });
+        DplDetector.hasMany(models.GaqDetector, { as: 'gaqAssociation', foreignKey: 'dplDetectorId' });
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/database/models/dpl/dpldetector.js]: applied

File: lib/server/services/dpl/DplDetectorsService.js (hunk 2 above)
Comment (xsalonx) [reply lib/server/services/dpl/DplDetectorsService.js]: moved to DataPassesService

File: lib/server/controllers/dataPasses.controller.js [hunk 4]
```diff
@@ -85,7 +85,36 @@ const setGaqDetectors = async (req, res) => {
...
+        req,
+        res,
+    );
+    if (validatedDTO) {
+        try {
+            const {
+                dataPassId,
+                runNumber,
+            } = validatedDTO.query;
+
```
Comment (martinboulais) [inline lib/server/controllers/dataPasses.controller.js:107]: Same line

**Review by martinboulais — APPROVED** [review]

**Pull request merged**